elastic/kibana
# [CTI] Adds Threat Intel dashboard links to Overview page

## Summary

This change adds Threat Intel dashboard links to the Overview page.

screenshot
<img width="1581" alt="Screen Shot 2021-05-27 at 6 58 40 PM" src="https://user-images.githubusercontent.com/18617331/119907153-aea5f280-bf1d-11eb-898b-df3ecfb421e9.png">

### notes
- currently the link panel does not show up if there are no events for the selected time range. This is to ensure that the user does not see a prompt to enable the threat intel module if they have the module enabled already, and has selected a time range with no events present.   
- ~~currently the link panel does not take the user to the Dashboard with the selected date range on the Overview page.~~ Dashboard links now carry over the time range from the securitySolution to the dashboard app. To enable this feature, the dashboard plugin had to be added to the security solution. 
- currently the link panel does not show up if there are enrichment events without a dashboard setup. (no links means no panel - we don't have a view that shows numbers only)

### To add dashboards with data

- start elasticsearch and kibana
- run `cd beats && git checkout master && git fetch && git pull && cd x-pack/filebeat`
- edit `beats/x-pack/filebeat/filebeat.yml` 
   - to include your elasticsearch username and password under `output.elasticsearch`
   - to include kibana username and password under `setup.kibana`

example
```
// beats/x-pack/filebeat/filebeat.yml
output.elasticsearch:
  hosts: ["localhost:9200"]
  username: "ece"
  password: "testtest"

setup.kibana:  
  host: "localhost:5601"
  username: "ece"
  password: "testtest"
```
 - run `mage build` to build filebeat
- run `./filebeat modules enable threatintel` and edit `modules.d/threatintel.yml` to ensure that all values for `enabled` are `true`
 - run  `./filebeat setup -E setup.dashboards.directory=build/kibana` to setup dashboards (while kibana is running)
 - run `./filebeat -e` to start filebeat 
 - after a few events are generated, you can stop filebeat. 

### Checklist

Delete any items that are not applicable to this PR.


Additional States

threat intel module disabled 
<img width="570" alt="Screen Shot 2021-06-22 at 2 21 52 PM" src="https://user-images.githubusercontent.com/18617331/122984488-0e4fbc00-d36b-11eb-87ae-d8b1bed19291.png">

dashboard module disabled and there are events for the selected time period
<img width="561" alt="Screen Shot 2021-06-22 at 11 49 44 AM" src="https://user-images.githubusercontent.com/18617331/122984485-0db72580-d36b-11eb-9808-07d9fab330b3.png">

dashboards are disabled and there are no events for selected time period
<img width="581" alt="Screen Shot 2021-06-22 at 11 49 26 AM" src="https://user-images.githubusercontent.com/18617331/122984492-0ee85280-d36b-11eb-8f31-664ec531ef88.png">

modules are enabled but there are no events for selected time period
<img width="569" alt="Screen Shot 2021-06-22 at 2 49 43 PM" src="https://user-images.githubusercontent.com/18617331/122984491-0e4fbc00-d36b-11eb-8960-4cd1be20de6a.png">

modules are enabled and there are events
<img width="587" alt="Screen Shot 2021-06-22 at 2 48 18 PM" src="https://user-images.githubusercontent.com/18617331/122984490-0e4fbc00-d36b-11eb-82d4-bb68beb870ca.png">
